src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/ElasticsearchUserCaseAccessFilterFactory.java

```diff
@@ -0,0 +1,28 @@
+package uk.gov.hmcts.ccd.domain.service.search.elasticsearch;
+
+import java.util.Optional;
+
+import static uk.gov.hmcts.ccd.domain.service.search.filter.CaseSearchFilter.CASE_ID;
+
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.domain.service.common.CaseAccessService;
+import uk.gov.hmcts.ccd.domain.service.search.filter.CaseFilterFactory;
+import uk.gov.hmcts.ccd.domain.service.search.filter.UserAccessFilter;
+
+@Component
+public class ElasticsearchUserCaseAccessFilterFactory extends UserAccessFilter implements CaseFilterFactory<QueryBuilder> {
```

Comment:
I can't see the benefit of having a UserAccessFilter superclass. I think we could merge it into ElasticsearchUserCaseAccessFilterFactory.

Reply:
what about changing to
```public class UserAccessQueryFilter implements ElasticsearchQueryFilter {

Reply:
same for the other filter

Reply:
it creates a filter that's why chose factory. Create looked better than applyFilter etc. Happy to change though.